nexus/db-queries/src/db/datastore/switch_port.rs

```diff
@@ -312,319 +316,2021 @@ impl DataStore {
+        let err = OptionalError::new();
+
+        self.transaction_retry_wrapper(
+            "switch_port_configuration_interface_bgp_peer_add",
+        )
+            .transaction(&conn, |conn| {
+                let parent_configuration = configuration.clone();
+                let new_settings = bgp_peer.clone();
+                let err = err.clone();
+
+                async move {
+                    // resolve id of port_settings record
+                    let port_settings_id = switch_port_configuration_id(&conn, parent_configuration.clone())
+                        .await
+                        .map_err(|e: diesel::result::Error| {
+                            match e {
+                                diesel::result::Error::NotFound => {
+                                    err.bail(
+                                        Error::non_resourcetype_not_found(
+                                            format!("unable to lookup configuration with identifier {parent_configuration}")
+                                        )
+                                    )
+                                },
+                                _ => {
+                                    err.bail(Error::internal_error(
+                                        "error while looking up configuration for interface bgp peer"
+                                    ))
+                                },
+                            }
+                        })?;
+
+                    // resolve id of referenced bgp configuratino
+                    let bgp_config_id = match new_settings.bgp_config {
+
+                        NameOrId::Id(id) => {
+                            // verify id is valid
+                            bgp_config::table
+                                .filter(bgp_config::time_deleted.is_null())
+                                .filter(bgp_config::id.eq(id))
+                                .select(bgp_config::id)
+                                .limit(1)
+                                .first_async::<Uuid>(&conn)
+                                .await
+                                .map_err(|e: diesel::result::Error| {
+                                    match e {
+                                        diesel::result::Error::NotFound => {
+                                            err.bail(Error::not_found_by_id(ResourceType::BgpConfig, &id))
+                                        },
+                                        _ => {
+                                            let message = "error while looking up bgp config for bgp peer";
+                                            error!(opctx.log, "{message}"; "error" => ?e);
+                                            err.bail(Error::internal_error(message))
+                                        },
+                                    }
+                                })
+                        },
```

Comment:
Do we allow renaming of resources today? It is my understanding that Name and ID are not changeable.

Reply:
ID is immutable, yeah -- I had meant to select the `NamrOrId::Name` match branch at the time.

My thinking is in case an update endpoint does crop up in future -- e.g., not all project scoped resources have update endpoints, but the resource name is almost always mutable when such an endopint exists.